test/Rakefile

```diff
@@ -64,12 +64,13 @@ namespace :compose do
   task :teardown_master do
     sh "docker-compose stop api mongodb"
     sh "docker-compose rm --force api mongodb"
+    sh "kontena master remove --force compose-e2e"
   end
 
   task :teardown_grid do
     sh "docker-compose stop agent"
     sh "docker-compose rm --force agent"
-    sh "kontena node rm --force moby"
+    sh "kontena node rm --force $HOSTNAME"
```

Comment:
There are multiple ways to run the tests, so not always you should remove the host where you are running the tests (and rake tasks) from. Maybe just remove all the nodes that are reported by `kontena node ls`?